Detections/SecurityEvent/password_never_expires.yaml

```diff
@@ -56,27 +56,33 @@ query: |
   | extend SubjectUserSid = tostring(EventData.SubjectUserSid)
   | extend AccountType=case(SubjectAccount endswith "$" or SubjectUserSid in ("S-1-5-18", "S-1-5-19", "S-1-5-20"), "Machine", isempty(SubjectUserSid), "", "User")
   | project StartTime = TimeGenerated, EventID, Activity, Computer, TargetAccount, TargetSid, AccountType, UserAccountControl, Value_2089, Value_2050, Value_2082, SubjectAccount
-  | extend timestamp = StartTime, AccountCustomEntity = TargetAccount, HostCustomEntity = Computer
+  | extend timestamp = StartTime
+  | extend HostName = tostring(split(Computer, ".")[0]), DomainIndex = toint(indexof(Computer, '.'))
+  | extend HostNameDomain = iff(DomainIndex != -1, substring(Computer, DomainIndex + 1), Computer)
+  | extend AccountName = tostring(split(TargetAccount, "@")[0]), AccountUPNSuffix = tostring(split(TargetAccount, "@")[1])
   )
-
 entityMappings:
   - entityType: Account
     fieldMappings:
-      - identifier: FullName
-        columnName: AccountCustomEntity
+      - identifier: Name
```

Comment:
TargetAccount and SubjectAccount are actually build from separated entity types already.  This is a legacy method.  We want Name and NTDomain as this is not a UPN.  You can get what you need by the following:
| extend TargetUserName = tostring(EventData.TargetUserName), TargetDomainName = tostring(EventData.TargetDomainName)

Then entity map these to Name and NTDomain.

Also, please map another entity to Account for SubjectUserName and SubjectDomainName.